packages/flame/lib/src/components/mixins/has_hitboxes.dart

```diff
@@ -1,12 +1,40 @@
 import 'dart:collection';
-import 'dart:ui';
 
+import 'package:flutter/cupertino.dart';
+
+import '../../../components.dart';
 import '../../../extensions.dart';
-import '../../geometry/shape.dart';
-import '../position_component.dart';
+import '../../../geometry.dart';
 
 mixin HasHitboxes on PositionComponent {
   final List<HitboxShape> _hitboxes = <HitboxShape>[];
+  final Aabb2 _aabb = Aabb2();
+  bool _validAabb = false;
+  final Vector2 _halfExtents = Vector2.zero();
+  final Matrix3 _rotationMatrix = Matrix3.zero();
+
+  @override
+  @mustCallSuper
+  void update(double dt) {
+    super.update(dt);
+    _validAabb = false;
+  }
+
+  Aabb2 get aabb {
+    if (_validAabb) {
+      return _aabb;
+    }
+    final size = scaledSize;
+    // This has +1 since a point on the edge of the bounding box is currently
+    // counted as inside.
+    _halfExtents.setValues(size.x + 1, size.y + 1);
+    _rotationMatrix.setRotationZ(absoluteAngle);
+    _aabb
+      ..setCenterAndHalfExtents(absoluteCenter, _halfExtents)
+      ..rotate(_rotationMatrix);
+    _validAabb = true;
+    return _aabb;
+  }
```

Comment:
Hmm, I think there is still a problem with the scale of all ancestors being ignored.

Also, note that in `PositionComponent` there is method `toAbsoluteRect()` which can probably be used here.

Reply:
We never came up with a solution for adding up scales right? It is documented that it doesn't work if you ancestors are scaled, but if you know how to solve it, it would of course be good to do so.
`toAbsoluteRect` is more expensive than this solution since it will have to project two vectors, and also we want an `Aabb2`, not a `Rect`, since we later can interact with the rest of the classes in `vector_math` with that.